File: cpp/39f_1850B_TenWordsOfWisdom.cpp
#include<bits/stdc++.h>

int main(){
    int t, n, a, b, ind, max_b;
    std::vector<int> ans;
    std::cin>>t;
    for (int j = 0; j < t; j++){
        std::cin>>n;
        ind = 0, max_b = 0;
        for(int j = 0; j < n; j++){
            std::cin>>a>>b;
            if(a <= 10){
                if(b > max_b){
                    max_b = b;
                    ind = j + 1;
                }
            }
        }
        ans.push_back(ind);
    }
    for(auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}

File: python/39f_1850B_TenWordsOfWisdom.py
def main():
    ans = []
    t = int(input())
    for _ in range(t):
        n = int(input())
        ind, max_b = 0, 0
        for i in range(n):
            a, b = input().split()
            a, b = int(a), int(b)
            if a <= 10:
                if b > max_b:
                    max_b = b
                    ind = i + 1
        ans.append(ind)
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()